test/earthen/clj_cb/bucket_test.clj

```diff
@@ -24,3 +33,181 @@
+    (is (= {:editions.2000 "1" :pages 12} (b/lookup-in (fx/bucket) (:name bigger-book) "editions.2000" "pages")))
+    (is (= {:editions.2001 "2" :publishers ["foo" "bar"]} (b/lookup-in (fx/bucket) (:name bigger-book) "editions.2001" "publishers")))
+    (is (= {:editions.2001 "2" :publishers ["foo" "bar"] :references<1>.item.label 99} (b/lookup-in (fx/bucket) (:name bigger-book) "editions.2001" "publishers" "references[1].item.label")))
+    (is (= {:exists nil} (b/lookup-in (fx/bucket) (:name bigger-book) "exists")))
+    (is (= {:exception "Path mismatch \"year.missing\" in bigger-living-clojure"} (b/lookup-in (fx/bucket) (:name bigger-book) "year.missing")))))
+
+(deftest query
+  (fx/authenticate "earthen" "earthen")
+  (b/create-primary-index (fx/bucket))
+  (dorun
+   (map #(b/replace!
+          (fx/bucket)
+          (str (:name bigger-book) "-" %)
+          (assoc bigger-book :name (str (:name bigger-book) "-" %)))
+        (range 10)))
+  (Thread/sleep 1000)
+  (is (= 10 (count (:rows (b/query (fx/bucket) "SELECT meta().id, editions FROM `earthen_test`")))))
+  (is (= 0 (count (:rows (b/query (fx/bucket) "SELECT * FROM `earthen_test` where name = \"not found\"")))))
+  (is (= 1 (count (:rows (b/query (fx/bucket) "SELECT * FROM `earthen_test` where name = \"bigger-living-clojure-9\"")))))
+  (let [result 
+        (b/query (fx/bucket) "SELECT meta().id FROM `earthen_test` where name = \"bigger-living-clojure-7\"")]
+    (is (= "success" (:status result)))
+    (is (= 1 (count (:rows result))))
+    (is (= "bigger-living-clojure-7" (:id (first (:rows result))))))
+  (let [result 
+        (b/query (fx/bucket) "SELECT meta().id FROM `earthen_test` LIMIT 3")]
+    (is (= "success" (:status result)))
+    (is (= 3 (count (:rows result))))
+    (is (= "bigger-living-clojure-0" (:id (first (:rows result)))))
+    (is (= "bigger-living-clojure-1" (:id (second (:rows result)))))
+    (is (= "bigger-living-clojure-2" (:id (nth (:rows result) 2)))))
+  (let [result 
+        (b/query (fx/bucket) "SELECT meta().id FROM `earthen_test` OFFSET 3 LIMIT 3")]
+    (is (= "success" (:status result)))
+    (is (= 3 (count (:rows result))))
+    (is (= "bigger-living-clojure-3" (:id (first (:rows result)))))
+    (is (= "bigger-living-clojure-4" (:id (second (:rows result)))))
+    (is (= "bigger-living-clojure-5" (:id (nth (:rows result) 2)))))
+  (let [result 
+        (b/query (fx/bucket) "SELECT meta().id FROM `earthen_test` OFFSET 9 LIMIT 1")]
+    (is (= "success" (:status result)))
+    (is (= 1 (count (:rows result))))
+    (is (= "bigger-living-clojure-9" (:id (first (:rows result)))))
+    )
+  (let [result 
+        (b/query (fx/bucket) "SELECT meta().id FROM `earthen_test` OFFSET 10 LIMIT 1")]
+    (is (= "success" (:status result)))
+    (is (= 0 (count (:rows result))))
+    (is (= 0 (.resultCount (:n1ql-metrics result))))))
+
+(deftest prepared-statement
```

Comment:
nice tests, would be nice if they can be translated on what are you tying to test.
it can be one test for intention on what are you trying to test.
(deftest prepared-statement
(testing "should parse from keyword"...)
(testing "should parse where clause"...), ...
or something similar.

Reply:
Sure, I will add some comments.